controllers/datadogagent/kubestatemetrics.go

```diff
@@ -0,0 +1,194 @@
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	kubeStateMetricsRBACName = "kube-state-metrics-core"
+	defaultKSMCoreConfigMap  = `
+---
+cluster_check: true
+init_config:
+instances:
+  - collectors:
+      - pods
+      - replicationcontrollers
+      - statefulsets
+      - nodes
+      - cronjobs
+      - jobs
+      - replicasets
+    telemetry: true
+  - collectors:
+      - deployments
+      - configmaps
+      - services
+      - endpoints
+      - daemonsets
+      - horizontalpodautoscalers
+      - limitranges
+      - resourcequotas
+      - secrets
+      - namespaces
+      - persistentvolumeclaims
+      - persistentvolumes
+    telemetry: true
+`
+	ksmCoreCheckName = "kubernetes_state_core.yaml"
+)
+
+func (r *Reconciler) manageKubeStateMetricsCore(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent) (reconcile.Result, error) {
+	if !isKSMCoreEnabled(dda.Spec.ClusterAgent) {
+		return reconcile.Result{}, nil
+	}
+	// Only create the default ConfigMap if the conf is not overridden
+	if dda.Spec.ClusterAgent.Config.KubeStateMetricsCoreConf != nil {
+		return reconcile.Result{}, nil
+	}
+	return r.manageConfigMap(logger, dda, datadoghqv1alpha1.GetKubeStateMetricsConfName(dda), buildKSMCoreConfigMap)
+}
+
+func buildKSMCoreConfigMap(dda *datadoghqv1alpha1.DatadogAgent) (*corev1.ConfigMap, error) {
+	configMap := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        datadoghqv1alpha1.GetKubeStateMetricsConfName(dda),
+			Namespace:   dda.Namespace,
+			Labels:      getDefaultLabels(dda, dda.Name, getAgentVersion(dda)),
+			Annotations: getDefaultAnnotations(dda),
+		},
+		Data: map[string]string{
+			ksmCoreCheckName: defaultKSMCoreConfigMap,
+		},
+	}
+	return configMap, nil
+}
+
+func (r *Reconciler) createKubeStateMetricsClusterRole(logger logr.Logger, dda *datadoghqv1alpha1.DatadogAgent, name, version string) (reconcile.Result, error) {
+	clusterRole := buildKubeStateMetricsCoreRBAC(dda, name, version)
+	logger.Info("createKubeStateMetricsClusterRole", "clusterRole.name", clusterRole.Name)
+	event := buildEventInfo(clusterRole.Name, clusterRole.Namespace, clusterRoleKind, datadog.CreationEvent)
+	r.recordEvent(dda, event)
+	return reconcile.Result{Requeue: true}, r.client.Create(context.TODO(), clusterRole)
+}
+
+// buildKubeStateMetricsCoreRBAC generates the cluster role required for the KSM informers to query
+// what is exposed as of the v2.0 https://github.com/kubernetes/kube-state-metrics/blob/release-2.0/examples/standard/cluster-role.yaml
+func buildKubeStateMetricsCoreRBAC(dda *datadoghqv1alpha1.DatadogAgent, name, version string) *rbacv1.ClusterRole {
+	clusterRole := &rbacv1.ClusterRole{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:      getDefaultLabels(dda, name, version),
+			Annotations: getDefaultAnnotations(dda),
+			Name:        name,
+		},
+	}
+
+	rbacRules := []rbacv1.PolicyRule{
```

Comment:
Is it worth adding something to the `DatadogAgent` CRD to customize this set of rules?

Reply:
That's what we discussed in standup last week, for now we keep it simple, but if someone asks that we make it customizable we can add the logic.